stubs/pyinstaller/METADATA.toml

```diff
@@ -1,5 +1,6 @@
 version = "5.7.*"
 requires = ["types-setuptools"]
+external_requires = ["modulegraph"]
```

Comment:
For now, for security reasons, any external dependencies will fail the stub uploader tests unless they're explicitly included in this whitelist here: https://github.com/typeshed-internal/stub_uploader/blob/bf783eb638a8aed4e1d0ba1143e3f03f399794a0/stub_uploader/metadata.py#L166

It's not ideal and we can think about ways to improve this after we've got the problems on the typeshed side up and running (plus in the medium term, we can just make that list longer) — but for now, we can't add `modulegraph` as a dependency, as it's not on that list

Reply:
Good to keep in mind! Eventually this PR will split-off stub changes (right now they're only included to test). So it shouldn't be an immediate issue.

Reply:
Having said that, it's interesting that the stub-uploader tests typeshed runs in CI still seem to be passing. Not sure whether that means I'm mistaken or whether it means the tests are buggy

Reply:
stub_uploader only looks at `requires` field. I think people disliked the suggestion to split the field up, so I kept it the same (there were also some subtleties around ordering and removing packages from typeshed and when to validate and stuff)

Reply:
@hauntsaninja Ah yes, that's something I meant to ask as well and slipped my mind: should it be the same or different field. And if the same, what's the logic for what needs to be installed? Probably the same as stubtest (a link to relevant code there would be appreciated). I already abstracted the logic to get non-type dependencies anyway.